### PR TITLE
Miken/new lru cache

### PR DESCRIPTION
**Pull request by digitalbuddha**

### CODE REVIEW

File: store/src/commonMain/kotlin/org/mobilenativefoundation/store/store5/impl/MemoryLruStore.kt [hunk 1]
```diff
@@ -2,118 +2,160 @@
...
+        } else {
+            node.value = value
+            moveNodeToHead(node)
+            return true
+        }
 
-    init {
-        head.next = tail
-        tail.prev = head
+        trim()
```
Comment (matt-ramotar) [inline store/src/commonMain/kotlin/org/mobilenativefoundation/store/store5/impl/MemoryLruStore.kt:51]: I think this won't hit

**Comment by matt-ramotar** [comment]
Should we keep the mutex?

File: store/src/commonMain/kotlin/org/mobilenativefoundation/store/store5/impl/MemoryLruStore.kt (hunk 1 above)
Comment (digitalbuddha) [reply store/src/commonMain/kotlin/org/mobilenativefoundation/store/store5/impl/MemoryLruStore.kt:51]: ?

**Comment by digitalbuddha** [comment]
I'll add mutex's back tomorrow

File: store/src/commonMain/kotlin/org/mobilenativefoundation/store/store5/impl/MemoryLruStore.kt (hunk 1 above)
Comment (matt-ramotar) [reply store/src/commonMain/kotlin/org/mobilenativefoundation/store/store5/impl/MemoryLruStore.kt:51]: isn't `trim()` unreachable?

**Pull request closed without merging**